### DISCUSSION TITLE
Why is my colour chart not detected?

### DISCUSSION THREAD

**Discussion by knorden** [colour-science/colour-checker-detection#29] [Q&A]
Hi @KelSolaar, thanks for providing this awesome resource! I'm trying to adapt this into my image processing flow to color correct images I will use for a research project. However, in the image I'm trialing the color chart appears to not be detected. I'm using your example code, which worked fine for another image of mine - so I'm not sure what the problem could be. The color chart I'm using is a version nano, which has slightly more rectangular squares - could this be the issue? Or does it have something to do with the lighting? Would appreciate any thoughts! (I'm attaching the type of image I'm working with - the original would be png but too large to attach here)
![image](https://user-images.githubusercontent.com/85402827/120895135-9e5ed900-c5e9-11eb-8ae5-f553dcca678a.jpg)

**Comment by KelSolaar** [comment]
Hi @knorden,

I think that the segmentation is failing because the edges are very fuzzy, the swatches are not square and the image is underexposed by the look of it:

![image](https://user-images.githubusercontent.com/99779/120920241-04468000-c712-11eb-924f-06e36341da28.png)

Do you have other cases with the same chart where it worked?

Cheers,

Thomas

**Reply by knorden** [reply]
No, it's only been working for me with the larger chart (with square swatches), so perhaps the format is the issue. I don't know whether the segmenting will also need the border of the chart itself? It's usually cut out from the images as well

**Reply by KelSolaar** [reply]
The chart ideally should have a geometry close to the ColorChecker24 Classic or Passport, not too big in the frame and not too small!
